gpf:
  ros__parameters:
    cropbox:
      enable: true
      min:
        x: -10.0
        y: -10.0
        z: -10.0
      max:
        x: 6.0
        y: 10.0
        z: 10.0

    downsampling:
      enable: true
      leaf_size:
        xy: 0.1
        z: 0.1

    supervoxel:
    # https://pcl.readthedocs.io/projects/tutorials/en/latest/supervoxel_clustering.html
    # https://www.sciencedirect.com/science/article/pii/S0924271618301370
      use_single_camera_transform: true
      voxel_resolution: 1.
      seed_resolution: 3.
      importance:
        spatial: 1.0
        color: 0.
      normal:
        enable: true
        importance: 4.0
        search_radius: 10.
      refinement:
        enable: true
        iterations: 5

    segmentation:
      lccp:
      # https://www.cv-foundation.org/openaccess/content_cvpr_2014/papers/Stein_Object_Partitioning_using_2014_CVPR_paper.pdf
        concavity_tolerance_threshold: 15.0
        smoothness_threshold: 0.25
        min_segment_size: 1
        k_factor: 0
        sanity_criterion: true
      cpc:
      # https://openaccess.thecvf.com/content_cvpr_2015/papers/Schoeler_Constrained_Planar_Cuts_2015_CVPR_paper.pdf
        max_cuts: 1000
        cutting_min_segments: 100
        cutting_min_score: 0.15
        local_constrain: true
        directed_cutting: true
        clean_cutting: true
        ransac_iterations: 2000

    sample_consensus:
      min_segment_size: 50
      plane:
        enable: false
        distance_threshold: 0.005
        probability: 0.99
        max_iterations: 1000
        min_inlier_proportion: 0.85
        merge_precision: 0.25
      sphere:
        enable: false
        distance_threshold: 0.005
        probability: 0.99
        max_iterations: 1000
        min_inlier_proportion: 0.7
        min_radius: 0.01
        max_radius: 2.
      cylinder:
        enable: true
        distance_threshold: 0.2
        probability: 0.99
        max_iterations: 1000
        min_inlier_proportion: 1.5
        min_radius: 0.05
        max_radius: 3.

    publish_markers: true
    visualise: false
    use_sim_time: false
